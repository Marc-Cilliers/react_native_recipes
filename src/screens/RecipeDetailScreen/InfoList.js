import React from 'react';
import {StyleSheet, View, Text} from 'react-native';

type Props = {
  title: string,
  content: Array<string>,
  numbered: boolean,
};

const InfoList = ({title, content, numbered}: Props): Object => {
  return (
    <View style={styles.container}>
      <Text style={styles.header}>{title}</Text>
      <View style={styles.seperator} />

      {content.map((c, i) => (
        <View key={i} style={styles.row}>
          {numbered && <Text style={styles.number}>{i + 1}.)</Text>}
          <Text style={styles.content}>
            {!numbered && 'â€¢'} {c}
          </Text>
        </View>
      ))}
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'column',
    marginTop: 20,
  },
  point: {
    textAlignVertical: 'bottom',
    fontSize: 5,
    marginRight: 10,
  },
  number: {
    fontSize: 15,
    marginRight: 20,
    width: 20,
    // width: 30,
  },
  row: {
    flexShrink: 1,
    marginLeft: 5,
    flexDirection: 'row',
    marginBottom: 5,
    justifyContent: 'flex-start',
    alignItems: 'flex-start',
  },
  content: {
    flexShrink: 1,
    fontSize: 16,
  },
  header: {
    fontSize: 25,
    fontWeight: '700',
  },
  seperator: {
    height: 1,
    marginBottom: 20,
    width: '90%',
    backgroundColor: 'grey',
  },
});

export default InfoList;
