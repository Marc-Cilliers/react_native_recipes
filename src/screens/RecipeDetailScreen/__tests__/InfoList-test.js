import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import InfoList from '../InfoList';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <InfoList
        title={'Ingredients'}
        content={[
          {ingredient: 'peaches', amount: '4'},
          {ingredient: 'butter', amount: '2 tbsp'},
          {ingredient: 'cinnamon', amount: '1 tsp'},
          {ingredient: 'pecans', amount: '120g'},
          {ingredient: 'rolled oats', amount: '30g'},
          {ingredient: 'plain Greek yoghurt', amount: '4 tbsp'},
        ]}
        numbered={false}
      />,
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
