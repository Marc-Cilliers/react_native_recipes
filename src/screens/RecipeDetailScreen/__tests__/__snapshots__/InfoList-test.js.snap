// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "column",
      "marginTop": 20,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 25,
        "fontWeight": "700",
      }
    }
  >
    Ingredients
  </Text>
  <View
    style={
      Object {
        "backgroundColor": "grey",
        "height": 1,
        "marginBottom": 20,
        "width": "90%",
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "flexDirection": "row",
        "flexShrink": 1,
        "justifyContent": "flex-start",
        "marginBottom": 5,
        "marginLeft": 5,
      }
    }
  >
    <Text
      style={
        Object {
          "flexShrink": 1,
          "fontSize": 16,
        }
      }
    >
      •
       
      Item 1
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "flexDirection": "row",
        "flexShrink": 1,
        "justifyContent": "flex-start",
        "marginBottom": 5,
        "marginLeft": 5,
      }
    }
  >
    <Text
      style={
        Object {
          "flexShrink": 1,
          "fontSize": 16,
        }
      }
    >
      •
       
      Item 2
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "flexDirection": "row",
        "flexShrink": 1,
        "justifyContent": "flex-start",
        "marginBottom": 5,
        "marginLeft": 5,
      }
    }
  >
    <Text
      style={
        Object {
          "flexShrink": 1,
          "fontSize": 16,
        }
      }
    >
      •
       
      Item 3
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "flexDirection": "row",
        "flexShrink": 1,
        "justifyContent": "flex-start",
        "marginBottom": 5,
        "marginLeft": 5,
      }
    }
  >
    <Text
      style={
        Object {
          "flexShrink": 1,
          "fontSize": 16,
        }
      }
    >
      •
       
      Item 4
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "flexDirection": "row",
        "flexShrink": 1,
        "justifyContent": "flex-start",
        "marginBottom": 5,
        "marginLeft": 5,
      }
    }
  >
    <Text
      style={
        Object {
          "flexShrink": 1,
          "fontSize": 16,
        }
      }
    >
      •
       
      Item 5
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "flexDirection": "row",
        "flexShrink": 1,
        "justifyContent": "flex-start",
        "marginBottom": 5,
        "marginLeft": 5,
      }
    }
  >
    <Text
      style={
        Object {
          "flexShrink": 1,
          "fontSize": 16,
        }
      }
    >
      •
       
      Item 6
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "flexDirection": "row",
        "flexShrink": 1,
        "justifyContent": "flex-start",
        "marginBottom": 5,
        "marginLeft": 5,
      }
    }
  >
    <Text
      style={
        Object {
          "flexShrink": 1,
          "fontSize": 16,
        }
      }
    >
      •
       
      Item 7
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "flexDirection": "row",
        "flexShrink": 1,
        "justifyContent": "flex-start",
        "marginBottom": 5,
        "marginLeft": 5,
      }
    }
  >
    <Text
      style={
        Object {
          "flexShrink": 1,
          "fontSize": 16,
        }
      }
    >
      •
       
      Item 8
    </Text>
  </View>
</View>
`;
